homeassistant/components/bond/entity.py

```diff
@@ -65,18 +64,26 @@ def should_poll(self) -> bool:
         return False
 
     @property
-    def device_info(self) -> dict[str, Any] | None:
+    def device_info(self) -> DeviceInfo:
         """Get a an HA device representing this Bond controlled device."""
-        device_info = {
-            ATTR_NAME: self.name,
+        device_info: DeviceInfo = {
             "manufacturer": self._hub.make,
-            "identifiers": {(DOMAIN, self._hub.bond_id, self._device.device_id)},
-            "suggested_area": self._device.location,
-            "via_device": (DOMAIN, self._hub.bond_id),
+            "identifiers": {
```

Comment:
The dev docs aren't really clear that this isn't allowed for identifiers, but it does seem explicit for `connections`

https://developers.home-assistant.io/docs/device_registry_index/#device-properties

AFAICT, the only requirement for identifiers is they have to be unique and hashable

Reply:
I think it would be ok to limit it to a two string tuple. There should be no problem to concatenate the strings into a single string as needed for the second string item.

Reply:
We should probably update the docs as well.

Reply:
Device registry has identifiers as str two-tuples everywhere though. One approach to fixing this would be to relax that somehow, but I don't know if that's seen appropriate or what's the intent.

Reply:
> I think it would be ok to limit it to a two string tuple. There should be no problem to concatenate the strings into a single string as needed for the second string item.

Agreed, but I suppose just doing that would cause the newly structured identifiers not match their old saved device registry entries for the bond integration, some migration code would be needed.

Reply:
We haven't enforced two string tuple and I have seen tuples of length 1 and 3. 

I think that for typing, we should type what we do right now. And in another PR we can decide to restrict it to 2.

Note, it was officially meant to be 2.

Reply:
#49864 implements that, will adjust here once that's in. That won't be enough to make bond pass cleanly here as it has Optionals in the identifier str tuple, but decreases the need for the workaround.